src/libraries/System.Net.NameResolution/tests/FunctionalTests/GetHostAddressesTest.cs

```diff
@@ -184,13 +184,17 @@ public async Task DnsGetHostAddresses_PostCancelledToken_Throws()
             using var cts = new CancellationTokenSource();
 
             Task task = Dns.GetHostAddressesAsync(TestSettings.UncachedHost, cts.Token);
-
-            // This test might flake if the cancellation token takes too long to trigger:
-            // It's a race between the DNS server getting back to us and the cancellation processing.
             cts.Cancel();
 
-            OperationCanceledException oce = await Assert.ThrowsAnyAsync<OperationCanceledException>(() => task);
-            Assert.Equal(cts.Token, oce.CancellationToken);
+            // This test is nondeterministic, the cancellation may take too long to trigger:
+            // It's a race between the DNS server getting back to us and the cancellation processing.
+            // since the host does not exist, both cases throw an exception.
+            Exception ex = await Assert.ThrowsAnyAsync<Exception>(() => task);
+            if (ex is OperationCanceledException oce)
+            {
```

Comment:
I'm a bit late to the party, but the test will now no longer fail if the implementation lacks actual cancellation support, so it became mostly useless.

In theory, `[Collection(nameof(DisableParallelization))]` was here to somewhat mitigate the race. I wonder on what OS-es did this fail / succeed?

I see Ubuntu in kusto, which doesn't make sense to me, since the test is marked with `[ActiveIssue(***, TestPlatforms.AnyUnix)]` which AFAIK should disable the test on Linux.

Reply:
> I wonder if there is something specific about the OS-es/machines/queues where this is failing?

It is failing in jitstress, which may increase the chance of the cancelling thread losing the race.

Reply:
Hmm, I have fallen to another trap... I will revert that PR then.

We should be able to detect jitstress by looking at envvars, I will take a look

Reply:
It was super-confusing. We may consider repopening #33378, but it's very unlikely that we'll address in the near future, so probably it's better to just skip the test on Unix with `[PlatformSpecific]` + a comment.